micro-rdk/src/esp32/pin.rs

```diff
@@ -185,34 +185,43 @@ impl Esp32GPIOPin {
         self.interrupt_type.is_some()
     }
 
-    pub fn setup_interrupt(&mut self, intr_type: InterruptType) -> Result<(), BoardError> {
-        match &self.interrupt_type {
-            Some(existing_type) => {
-                if *existing_type == intr_type {
-                    return Ok(());
-                }
-            }
-            None => {
-                self.interrupt_type = Some(intr_type);
-            }
-        };
+    pub(crate) fn setup_interrupt(
+        &mut self,
+        intr_type: InterruptType,
+        cb: gpio_isr_t,
+        arg: *mut core::ffi::c_void,
+    ) -> Result<(), BoardError> {
+        if cb.is_none() {
+            return Err(BoardError::GpioPinOtherError(
+                self.pin as u32,
+                "interrupt callback cannot be `None`".to_string().into(),
+            ));
+        }
+
+        if self.pin == 36 || self.pin == 39 {
+            // When ADC or WiFi and Bluetooth with sleep mode enabled are powered on,
```

Comment:
while I follow the logic here I am against adding exclusion that affect only specifics model of esp32s. In theory the same board implementation is used for esp32s2/s3/c6 and that bug doesn't exists there. We usually try to avoid this type of exclusion leaving the user to figure out what went wrong. 
lastly since gpio36 & 39 are input only they will be excluded by the is_gpio_valid check in pin.rs.

Reply:
removed